source/hic_hal/target_config.h

```diff
@@ -55,7 +55,8 @@ typedef struct target_cfg {
     uint8_t erase_reset;            /*!< Reset after performing an erase */
 } target_cfg_t;
 
-extern const target_cfg_t target_device;
+extern target_cfg_t target_device;
+void set_target_device(uint32_t device);
```

Comment:
It can be non-const but you'll probably need to change it for all the targets. I would try building another target and see if you get compilation warnings.